src/anet.c

```diff
@@ -420,11 +423,15 @@ static int anetTcpGenericConnect(char *err, const char *addr, int port,
         /* Try to create the socket and to connect it.
          * If we fail in the socket() call, or on connect(), we retry with
          * the next entry in servinfo. */
-        if ((s = socket(p->ai_family,p->ai_socktype,p->ai_protocol)) == -1)
+#ifdef SOCK_NONBLOCK
+        if (flags & ANET_CONNECT_NONBLOCK) p->ai_socktype |= SOCK_NONBLOCK;
+#endif
+        if ((s = anetCreateSocket(err,p->ai_family,p->ai_socktype,p->ai_protocol,SO_REUSEADDR)) == -1)
             continue;
-        if (anetSetReuseAddr(err,s) == ANET_ERR) goto error;
+#ifndef SOCK_NONBLOCK
```

Comment:
maybe anetCreateSocket should handle anetNonBlock as well, like it handles anetSetReuseAddr?